scripts/build_shapes.py

```diff
@@ -120,10 +128,51 @@ def get_GADM_layer(country_list, layer_id, geo_crs, update=False, outlogging=Fal
             three_2_two_digits_country(twoD_c) for twoD_c in geodf_temp["GID_0"]
         ]
 
+        # GID_0 may have some exotic values, "COUNTRY" column may be used instead
+        geodf_temp["GID_0"] = geodf_temp.apply(
+            lambda x: restore_country_code_by_name(x, country_code=country_code), axis=1
+        )
+
+        # "not found" hardcoded according to country_converter conventions
+        geodf_temp.drop(
+            geodf_temp[geodf_temp["GID_0"] == "not found"].index, inplace=True
+        )
+
         # create a subindex column that is useful
         # in the GADM processing of sub-national zones
         geodf_temp["GADM_ID"] = geodf_temp[f"GID_{layer_id}"]
 
+        if layer_id >= 1:
+            available_gadm_codes = geodf_temp["GADM_ID"].unique()
+            code_three_digits = two_2_three_digits_country(country_code)
+
+            # normally the GADM code starts the ISO3
+            non_std_gadm_codes = [
+                w for w in available_gadm_codes if not w.startswith(code_three_digits)
+            ]
+
+            if len(non_std_gadm_codes) > 0:
+
+                # a "minimalistic" approach outputs a single file (~200kb for CN + IN)
+                d = {
+                    "country_code": country_code,
+                    "non_standard_gadm_codes": non_std_gadm_codes,
+                }
+                df = pd.DataFrame(data=d)
```

Comment:
This code is super useful for debugging, but for more than 1 country it would need some changes.
For example, we would need to pack all non-contended zones in a single file and add the path to the output file by snakemake to make it snakemake-compliant.
I think it is super useful for debugging, but for general use may create noise.
As this part may be moved into another function, we can add an argument to the function to state whether to save this output or not.
By default, I wouldn't like to have this output.

What do you think?

Reply:
Absolutely agree :) Probably, we could move this part to a second step